tck/src/main/java/ee/jakarta/tck/data/common/cdi/Directory.java

```diff
@@ -31,5 +35,12 @@ public interface Directory extends DataRepository<Person, Long> {
     //Self referencing provider so that this repository can be used for both WEB and CORE tests
     public static final String PERSON_PROVIDER = "PERSON_PROVIDER";
     
-    List<String> findFirstNameByIdInOrderByAgeDesc(List<Long> ids);
+    List<String> findLastNameByIdInOrderByAgeDesc(List<Long> ids);
```

Comment:
I think this is still wrong, isn't it? The problem wasn't that `First` was ambiguous, the problem was that QbMN simply does not have a projection operator.

Reply:
Sorry had to re-read the spec to understand completely.  Updated the test so it should follow the query by method name BNF